### DISCUSSION TITLE
Running rules multiple times with different options

### DISCUSSION THREAD

**Discussion by thomasparsons** [eslint/eslint#14688] [General]
I've been looking for this and struggled to find an answer

I imagine the answer right now is no, as I assume it's mighty complicated, but I do have a business case for needing to run the same rule multiple times with different options

We currently use `react/forbid-elements` to stop engineers from using HTML elements that we have internal components for, like so:

```
'react/forbid-elements': [2, {forbid: ['componentA']}],
``` 

What I need to be able to do is add this rule again like so:

```
'react/forbid-elements': [1, {forbid: ['componentB']}],
```

as some should error and some should warn

---

How could I go about doing this?

**Comment by nzakas** [comment]
This isn’t possible. Each rule can have exactly one configuration.

**Reply by thomasparsons** [reply]
Well, I could fork the rule, rename it and then use it, so it is _possible_

what I'm looking for is a slightly easier way of doing that;

Could you tell me, where does the actual name of the eslint rule come from - I assume it's the `name` value in the package.json of the rule itself? - as I assume, its the name of the rule that forces the singular use of each rule, if hypothetically we could rename the rule, or alias it, then that _could_ mean we could use the rule twice right?

**Reply by nzakas** [reply]
The name of the rule is determined in the plugin object. We combine the plugin name with the rule name as specified by the exported rules object. See https://eslint.org/docs/developer-guide/working-with-plugins

So yes, if you duplicated the rule and gave it a different name, you could run it with two different configs. There’s just not an easy way to do that in the current config system (the upcoming new config system makes this very easy)

**Reply by thomasparsons** [reply]
@nzakas thanks for this, that's really interesting

I would love to hear more about updates to eslint, is there anywhere this is shown/documented?

I am a big advocate for eslint, and would always be open to beta testing things if needed!

**Reply by nzakas** [reply]
All of the big changes go through the RFC process, so the rfcs repo is a good place to keep an eye on

**Comment by mdjermanovic** [comment]
One workaround is to make a plugin that re-exports rules from another plugin (or plugins), and that plugin could export the same rule multiple times under different names.

An example of a plugin that exports rules from other plugins: https://github.com/mysticatea/eslint-plugin

**Reply by thomasparsons** [reply]
that's an interesting idea! thanks @mdjermanovic

**Comment by phoenixstudiodz** [comment]
This is not an answer to your question rather an observation about the syntax.
```Javascript
{
    'react/forbid-elements': [2, {forbid: ['componentA']}],
    'react/forbid-elements': [1, {forbid: ['componentB']}]
}
```
This is equivalent to 

```Javascript
{
    a: 1,
    a: 2
}
```
This should throw an error in some programming languages or at least a warning because a key can not have two values.
Usually the first value is ignored and the value of `a` is `2`

Same apply to your rules, only the last one (key/rule name) will be visible to `eslint`.
If you are experiencing that, this has no relation to `eslint` rather it is how objects are treated in `Javascript`
